dtls.c

```diff
@@ -1841,8 +1841,36 @@ dtls_verify_peer(dtls_context_t *ctx,
 }
 
 #ifdef DTLS_ECC
+/*
+ * Assumes that data_len is at least 1 */
+static size_t
+dtls_asn1_len(uint8 **data, size_t *data_len)
+{
+  size_t len = 0;
+
+  if ((**data) & 0x80) {
+    size_t octets = (**data) & 0x7f;
+    (*data)++;
+    (*data_len)--;
+    while (octets) {
+      len = (len << 8) + (**data);
+      (*data)++;
+      if (*data_len == 0)
+        return -1ULL;
```

Comment:
You need a cast to `size_t` here because size_t is sometimes defined as `unsigned long` (especially on platforms for embedded devices).

Reply:
Updated.